Create 2021-12-18_DAW_Adventures_IN_WDIO.md (#7824)
webdriverio/webdriverio: website/blog/2021-12-18_DAW_Adventures_IN_WDIO.md

# Adventures in WDIO with Paul Grossman the DarkArtsWizard 
http://github.com/darkartswizard

> 

Reviewer: Can you prefix the file similar to other blog posts, e.g.:
```
---
title: WebdriverIO, meet Serenity/JS
author: Jan Molak
authorURL: https://www.linkedin.com/in/janmolak/
authorImageURL: https://avatars.githubusercontent.com/u/1089173?s=400&v=4
---
```

Author: Changes made as requested

---
title: Adventures in WebdriverIO (Blog post #1)
author: Paul Grossman DarkArtsWizard
authorURL: http://github.com/darkartswizard
authorImageURL: https://testguild.com/wp-content/uploads/2016/05/PaulGrossmanHeadShot.jpg
---
 
## The beginning...
 
**What makes you so special?**

I have been a test automation framework architect since 2001.

It started with my impossible automation script resulting in a perfect game of Pogo's Turbo 21.

Just for fun.

No one was there to tell me you can't automate a JDK game without any xPath or CSS element locators.

I just did it.

> 
> 
> 

Reviewer: ```suggestion

```

My first official project was intended to be the last gasp of a defunct 
project using the leading tool of the day: **Mercury Interactive's WinRunner**.

> After a week of record / playback 
> I was able to pull off a demo for the Dev and managers with a single end-to-end script!
> 
> 

Reviewer: ```suggestion
After a week of record / playback I was able to pull off a demo for the Dev and managers with a single end-to-end script!

```

![Imagine a code version of a game of Jenga! ](https://content.instructables.com/ORIG/FEZ/3ABS/FINO9YOI/FEZ3ABSFINO9YOI.jpg)

Any mild wind would cause it to break.

The next day I began to learn of the advantage of **Modular Code Design** and other best practices.

> If you want to hear the whole story of a perfectly timed magic trick listen to my first podcast with Joe Colantonio: www.testguild.com/103

Reviewer: ```suggestion
If you want to hear the whole story of a perfectly timed magic trick listen to [my first podcast](https://testguild.com/podcast/automation/103-five-test-automation-secrets-paul-grossman/) with Joe Colantonio.
```

I was able to show a $2 million ROI on my first outing. This allowed me to move from part-time automator to full-time framework architect.

**Long story, short**: From there I moved on to projects using **AutoItScript**, **MicroFocus QTP/UFT with VbScript**, various flavors of **Selenium in Java**, one with **Spring Boot.**
I also presented at automation conferences **always with a live demo**.

## Fast forward to 2019 pre-Covid 

Just to keep my mind sharp during an extended five month period of unemployment, 
I dove at the chance to pickup a deep discount subscription for every programming laguage supported by **IntelliJ's JetBrains**.
This meant my programming languages were no longer limited. I studied various frameworks including **Serenity**, **Ruby On Rails** and **CodeceptJS**.

> I then wrote the **Magic Object Model plugin** for **TestProject.io** 

Reviewer: ```suggestion
I then wrote the **Magic Object Model plugin** for **TestProject.io**.
```

That led me to my weekends being taken up by **testRigor**, which is an entirely different beast for Manual Testers and a story for another day. 

> 
> 

Reviewer: ```suggestion

```

## The Story So far...

Five months ago I accepted the daughting task to fill a vacant lead position to drive six different test automation projects.
One had massive longjevity. 
two were just getting starting. 
Three more were yet to get underway.

All had one thing in common: **WebdriverIO with JavaScript**

The prior Project Lead had intended to have a common framework called **One-Test** that could be leveraged by all other web projects.

**"One-Test to rule them all!****

I jumped in with both feet into the deep end working with the best supporting team of my career.
 
 

> And today I began to read the **WebDriverIO online instruction manual** from the beginning to figure out how all this stuff actually connects.

Reviewer: Can you reference the online instruction manual?

I know, I probably should have done that on Day One, but I really wanted to add ASCII Color coded logging to the Console.log first.

```diff
+ Clicked on the  
- "OK" button
+ on the 
- "Primary Address"
+ page. 

> ``` 

Reviewer: I don't really understand how this is connected to the previous text.
